src/clang_complete.h

```diff
@@ -145,4 +148,8 @@ struct ClangCompleteManager {
   // Parse requests. The path may already be parsed, in which case it should be
   // reparsed.
   ThreadedQueue<PreloadRequest> preload_requests_;
+  int* config_options[SIZE_OF_OPTIONS] = {&kMaxPreloadedSessions,
+                                          &kMaxCompletionSessions};
+  // Changing dynamically the options defined in config_options
+  void ConfigurableOptions(int* config_options[], int option, int value);
```

Comment:
Why is `config_options` an argument of the function *and* a member of the class? This doesn't really make sense. 
Was the intended usage to have specified, at the call site, the index and value of the option? If so you can just drop the first function argument, but then `ConfigurableOptions(0,1)` is far from obvious.

Reply:
Because i am developing on C projects lately i forget the namespace thing and made a bad design due to that.